crates/cli/src/commands/format.rs

```diff
@@ -0,0 +1,176 @@
+use crate::{
+    resolver::{resolve_from_cwd, Source},
+    ux::format_diff,
+};
+use anyhow::{anyhow, ensure, Context, Result};
+use biome_grit_formatter::context::GritFormatOptions;
+use clap::Args;
+use colored::Colorize;
+use marzano_gritmodule::{config::ResolvedGritDefinition, parser::PatternFileExt};
+use serde::Serialize;
+
+#[derive(Args, Debug, Serialize, Clone)]
+pub struct FormatArgs {
+    /// Write formats to file instead of just showing them
+    #[clap(long)]
+    pub write: bool,
+}
+
+pub async fn run_format(arg: &FormatArgs) -> Result<()> {
+    let (mut resolved, _) = resolve_from_cwd(&Source::Local).await?;
+    // sort to have consistent output for tests
+    resolved.sort();
+
+    let file_path_to_resolved = group_resolved_patterns_by_group(resolved);
+    for (file_path, resovled_patterns) in file_path_to_resolved {
```

Comment:
This can be easily executed in parallel, however the tests that verify stdout will fail due to inconsistencies in the output

Reply:
Instead of changing the input to make tests happy, another option is to sort the test output by line so it is stabilized.

Reply:
done